packages/playground/src/util/cacheHandlers.ts

```diff
@@ -0,0 +1,48 @@
+import {
+  cloneElementInCode,
+  deleteElementInCode,
+  findElementInAST,
+} from './codeHandlers';
+
+const getDataFromCache = async (
+  file: string,
+  char: number,
+): Promise<{ body?: string; cache?: Cache; charCount?: number }> => {
+  try {
+    const cache = await caches.open('playground');
+    const cacheContent = await cache.match(`/controlled/${file}`);
+    const body = await cacheContent?.text();
+
+    console.log(char);
+    const clickedNode = findElementInAST(body || '', char);
+    if (!clickedNode) throw new Error('Element was not found in code');
+
+    const charCount = clickedNode.end - clickedNode.start;
+    return { body, cache, charCount };
+  } catch (err) {
```

Comment:
Never Swallow Exceptions
https://joefallon.net/2018/09/typescript-try-catch-finally-and-custom-errors/